x-pack/plugins/aiops/public/components/log_rate_analysis_results_table/use_columns.tsx

```diff
@@ -0,0 +1,357 @@
+import { useViewInDiscoverAction } from './use_view_in_discover_action';
+import { useViewInLogPatternAnalysisAction } from './use_view_in_log_pattern_analysis_action';
+import { useCopyToClipboardAction } from './use_copy_to_clipboard_action';
+import { MiniHistogram } from '../mini_histogram';
+
+const TRUNCATE_TEXT_LINES = 3;
+const ACTIONS_COLUMN_WIDTH = '60px';
+const NARROW_COLUMN_WIDTH = '120px';
+const UNIQUE_COLUMN_WIDTH = '40px';
+const NOT_AVAILABLE = '--';
+
+export const commonColumns = ['Log rate', 'Doc count', 'p-value', 'Impact', 'Actions'];
+export const significantItemColumns = ['Field name', 'Field value', ...commonColumns] as const;
+export const GROUPS_TABLE = 'groups';
+export const SIG_ITEMS_TABLE = 'significantItems';
+type TableType = typeof GROUPS_TABLE | typeof SIG_ITEMS_TABLE;
+type ColumnNames = typeof significantItemColumns[number] | 'unique';
+
+const logRateHelpMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTable.logRateColumnTooltip',
+  {
+    defaultMessage:
+      'A visual representation of the impact of the field on the message rate difference',
+  }
+);
+const groupLogRateHelpMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTableGroups.logRateColumnTooltip',
+  {
+    defaultMessage:
+      'A visual representation of the impact of the group on the message rate difference.',
+  }
+);
+const groupImpactMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTableGroups.impactLabelColumnTooltip',
+  {
+    defaultMessage: 'The level of impact of the group on the message rate difference',
+  }
+);
+const impactMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTable.impactLabelColumnTooltip',
+  {
+    defaultMessage: 'The level of impact of the field on the message rate difference.',
+  }
+);
+
+export const useColumns = (
+  tableType: TableType,
+  skippedColumns: string[],
+  searchQuery: estypes.QueryDslQueryContainer,
+  timeRangeMs: TimeRangeMs,
+  loading: boolean,
+  zeroDocsFallback: boolean,
+  barColorOverride?: string,
+  barHighlightColorOverride?: string,
+  isExpandedRow: boolean = false
+): Array<EuiBasicTableColumn<SignificantItem>> => {
+  const { data, uiSettings, fieldFormats, charts } = useAiopsAppContext();
+  const { dataView } = useDataSource();
+  const euiTheme = useEuiTheme();
+  const viewInDiscoverAction = useViewInDiscoverAction(dataView.id);
+  const viewInLogPatternAnalysisAction = useViewInLogPatternAnalysisAction(dataView.id);
+  const copyToClipBoardAction = useCopyToClipboardAction();
+
+  const isGroupsTable = tableType === GROUPS_TABLE;
+
+  const fieldStatsServices: FieldStatsServices = useMemo(() => {
+    return {
+      uiSettings,
+      dataViews: data.dataViews,
+      data,
+      fieldFormats,
+      charts,
+    };
+  }, [uiSettings, data, fieldFormats, charts]);
+
+  const columnsMap: Record<ColumnNames, EuiBasicTableColumn<SignificantItem>> = useMemo(
+    () => ({
+      ['Field name']: {
```

Comment:
Curious how would this work for internationalized versions 🤔 ? Right now the search logic for `columnNamesToReturn` is hardcoding `'Impact'` and that the search text will be in English.

Reply:
Oh I see, these are just to determine which columns to return from `significantItemColumns` and `commonColumns`. I'm not so sure if these 2 constants should be using the hardcoded english labels. Suggestion to use the form of unique column id instead and use an enum or a map to translate to i18n label strings, or to internationalize the strings in these constants and always (which would translate better when users type search terms as well).

Reply:
The labels come from the `name` property of the columns - which are internationalized. This PR only moved the columns to the hook to remove duplication.

Reply:
I might be missing something. The `name` column is internationalized but when you're searching in the input, it's only searching based on the English strings, which come form `uniqueItemNames`

 ```
  const filteredUniqueItemNames = useMemo(() => {
    return uniqueItemNames.filter(
      (d) => d.toLowerCase().indexOf(itemSearchText.toLowerCase()) !== -1
    );
  }, [itemSearchText, uniqueItemNames]);
```

The uniqueItemNames come from 
```
            uniqueItemNames={(groupResults ? commonColumns : significantItemColumns) as string[]}
```

and commonColumns and significantItemColumns are currently hardcoded English labels.
```
export const commonColumns = ['Log rate', 'Doc count', 'p-value', 'Impact', 'Actions'];
export const significantItemColumns = ['Field name', 'Field value', ...commonColumns] as const;
```

Reply:
Ah, yes - I misunderstood - I thought you were referring to the column names. Yes, good call. Will update 👍

Reply:
Ensured column names are internationalized now - in 72f120e9a39c579999d4b3aabcf548ff88d6fdf0